### ISSUE TITLE
`Invalid IAP assertion` for Cloud Run configured JIT Access application

### ISSUE THREAD

**Issue by rialg** [GoogleCloudPlatform/jit-access#73]
### Configuration

* version: `1.4.0` 
* scope: `Project`
 
### Description

After deploying the application following along the steps for Cloud Run in [this guide](https://cloud.google.com/architecture/manage-just-in-time-privileged-access-to-project#deploy_the_application), the error message `Invalid IAP assertion (HTTP 403: error)`  caused by [this exceptions](https://github.com/GoogleCloudPlatform/jit-access/blob/1.4.0/sources/src/main/java/com/google/solutions/jitaccess/web/IapRequestFilter.java#L124) appeared.

I tried disabling HTTPS forwarding and granting the `role/run.invoker` to the SA `service-[PROJECT-NUMBER]@gcp-sa-iap.iam.gserviceaccount.com` , as explained in the [documentation about IAP and Cloud Run](https://cloud.google.com/iap/docs/enabling-cloud-run#enabling)

**Comment by jpassing** [comment]
Was the error message `Expected audience does not match` not recorded in the logs? Because that would be a bug I'd like to fix.

(The HTTP responses intentionally don't contain any details that could be sensitive)

**Comment by rialg** [comment]
@jpassing Exactly, I had to add a log line in the catch block in order to see the error message.

**Comment by duxbuse** [comment]
I am also getting this issue 
`Invalid IAP assertion (HTTP 403: Forbidden)`

I am running inside a VPC. Is there some magic to let in ingress from the IAP into the vpc?

**Comment by jpassing** [comment]
Fixed the error handling in #80 and #93. If `IAP_BACKEND_SERVICE_ID` wasn't set correctly, the logs now show an error like 

```
Verifying IAP assertion failed. This might be because the IAP assertion was tampered with, or because it had the wrong audience (expected audience: /projects/xxx/global/backendServices/yyy).: Expected audience does not match
```

The fix is available in the `master` branch.
